File: src/main/java/com/qtech/bubbles/effect/ParalyzeEffect.java
package com.qtech.bubbles.effect;

import com.qtech.bubbles.common.effect.StatusEffect;
import com.qtech.bubbles.common.effect.StatusEffectInstance;
import com.qtech.bubbles.entity.Entity;
import com.qtech.bubbles.entity.player.PlayerEntity;
import com.qtech.utilities.python.builtins.ValueError;

public class ParalyzeEffect extends StatusEffect {
    public ParalyzeEffect() throws ValueError {
        super();
    }

    @Override
    protected boolean canExecute(Entity entity, StatusEffectInstance statusEffectInstance) {
        return false;
    }

    @Override
    public void onStart(StatusEffectInstance statusEffectInstance, Entity entity) {
        if (entity instanceof PlayerEntity) {
            entity.setMotionEnabled(false);
        }
    }

    @Override
    public void onStop(Entity entity) {
        if (entity instanceof PlayerEntity) {
            entity.setMotionEnabled(true);
        }
    }

    @Override
    protected void updateStrength() {
        // Do nothing.
    }
}

File: src/main/kotlin/com/qtech/bubbles/effect/ParalyzeEffect.kt
package com.qtech.bubbles.effect

import com.qtech.bubbles.common.effect.Effect
import com.qtech.bubbles.common.effect.EffectInstance
import com.qtech.bubbles.entity.Entity
import com.qtech.bubbles.entity.player.PlayerEntity

class ParalyzeEffect : Effect() {
    override fun canExecute(entity: Entity, effectInstance: EffectInstance): Boolean {
        return false
    }

    override fun onStart(effectInstance: EffectInstance, entity: Entity) {
        if (entity is PlayerEntity) {
            entity.isMotionEnabled = false
        }
    }

    override fun onStop(entity: Entity?) {
        if (entity is PlayerEntity) {
            entity.isMotionEnabled = true
        }
    }

    override fun updateStrength() {
        // Do nothing.
    }
}